src/main/java/com/onarandombox/MultiverseCore/destination/DestinationFactory.java

```diff
@@ -37,19 +43,103 @@ public DestinationFactory(MultiverseCore plugin) {
     }
 
     /**
-     * Gets a new destination from a string.
+     * Gets a new destination from a string and possibly for a player.
      * Returns a new InvalidDestination if the string could not be parsed.
      *
      * @param destination The destination in string format.
+     * @param player The player.
      *
      * @return A non-null MVDestination
      */
-    public MVDestination getDestination(String destination) {
+    public MVDestination getDestination(String destination, Player player) {
```

Comment:
If you look a little later in the diff, you'll see that's what I already did.  I didn't want to break the existing API.  The original method is just a shim that supports the original API.  It simply calls the new API with null for player and the behavior is the same as the old behavior and still keeps the code DRY.